src/qt/notifyznodewarning.cpp

```diff
@@ -2,47 +2,46 @@
 
 #include "evo/deterministicmns.h"
 #include "znode.h"
+#include "znodesync-interface.h"
 #include "chain.h"
 #include "znodeconfig.h"
 #include "znodeman.h"
+#include "warnings.h"
+#include "validation.h"
 
 
 #ifdef ENABLE_WALLET
 #include "walletmodel.h"
 #include "wallet/wallet.h"
 #endif
 
-#include <QMessageBox>
-#include <QAbstractButton>
-
 bool NotifyZnodeWarning::nConsidered = false;
 
 void NotifyZnodeWarning::notify()
 {
-    QMessageBox msg;
     const Consensus::Params& params = ::Params().GetConsensus();
     float numBlocksToEnforcement = params.DIP0003EnforcementHeight - chainActive.Tip()->nHeight;
     float minutesToEnforcement = numBlocksToEnforcement * (params.nPowTargetSpacingMTP / 60);
     float daysDecimal = minutesToEnforcement / 60 / 24;
     float daysToEnforcement = floor(daysDecimal);
     float hoursToEnforcement = floor((daysDecimal > 0 ? (daysDecimal - daysToEnforcement) : 0) * 24);
 
-    QString messageWarning = QString("WARNING: Legacy znodes detected. You should migrate to the new Znode layout before it becomes enforced (approximately %1 days and %2 hours). For details on how to migrate, go to https://zcoin.io/znode-migration")
-    .arg(QString::number((int)daysToEnforcement, 10))
-    .arg(QString::number((int)hoursToEnforcement, 10));
-    msg.setText(messageWarning);
-    msg.setIcon(QMessageBox::Warning);
-    msg.exec();
-    nConsidered = true;
+    std::string strWarning = strprintf(_("WARNING: Legacy znodes detected. You should migrate to the new Znode layout before it becomes enforced (approximately %i days and %i hours). For details on how to migrate, go to https://zcoin.io/znode-migration"),
+        (int)daysToEnforcement,
+        (int)hoursToEnforcement);
+
+    SetMiscWarning(strWarning);
+    AlertNotify(strWarning);
```

Comment:
This particular call runs a program specified with `-alertnotify` argument and passes an argument of `strWarning` to it. I doubt we need it.

You should just call
```
uiInterface.NotifyAlertChanged();
```

Reply:
Done